File: api/create_conversation.py
from http.server import BaseHTTPRequestHandler
import json
import requests
import os

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Tavus API endpoint
        url = "https://api.tavus.io/v2/conversations"

        payload = {
            "avatar_id": os.environ["r9bfd10a3e41"],
            "input": {}
        }

        headers = {
            "Authorization": f"Bearer {os.environ['7c055a37d70c46719aaddaa9c34bc638']}",
            "Content-Type": "application/json"
        }

        response = requests.post(url, json=payload, headers=headers)
        data = response.json()

        # Return JSON to the browser
        self.send_response(200)
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(json.dumps(data).encode())

File: api/create_conversation.js
export default async function handler(req, res) {
  try {
    // 1. Check if variables exist
    if (!process.env.TAVUS_API_KEY || !process.env.AVATAR_ID) {
      return res.status(500).json({ error: "Missing Environment Variables in Vercel Settings" });
    }

    const url = "https://api.tavus.io/v2/conversations";

    // 2. Use 'replica_id' instead of 'avatar_id'
    const payload = {
      replica_id: process.env.AVATAR_ID,
      input: {}
    };

    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${process.env.TAVUS_API_KEY}`,
        "Content-Type": "application/json"
      },
      body: JSON.stringify(payload)
    });

    const data = await response.json();
    return res.status(200).json(data);

  } catch (err) {
    console.error("Server error:", err);
    return res.status(500).json({ error: "Server crashed", details: err.toString() });
  }
}
